src/app/CommandSender.cpp

```diff
@@ -563,15 +563,7 @@ CHIP_ERROR CommandSender::AddRequestData(const CommandPathParams & aCommandPath,
     ReturnErrorOnFailure(PrepareCommand(aCommandPath, prepareCommandParams));
     TLV::TLVWriter * writer = GetCommandDataIBTLVWriter();
     VerifyOrReturnError(writer != nullptr, CHIP_ERROR_INCORRECT_STATE);
-    if (mExchangeCtx.Get() && mExchangeCtx.Get()->HasSessionHandle())
```

Comment:
This is ok to not include a fabric index? i.e. what's CommandSender used for vs CommandHandlerImpl?

Reply:
CommandSender is used for client-to-server (request) commands, which per spec are required to not include a fabric index right?

CommandHandlerImpl is used for server-to-client (response) commands.

Reply:
Oh sweet jesus why are they both under `app`? 😬🤦‍♂️

I had thought client-to-server generally went under controller?

Does it make sense to add a note to `CommandSender` that it's intended for request commands, whereas `CommandHandler` for response?

(Feel free to resolve this either way, will defer to you)

Reply:
Yeah, better docs would be good.  And apps can send commands (e.g. a light switch!)